akka-typed/src/main/scala/akka/typed/internal/ActorCell.scala

```diff
@@ -315,10 +312,39 @@ private[typed] class ActorCell[T](
          * activation token again.
          */
         val again = unsafe.getAndAddInt(this, statusOffset, 1)
-        if (activations(again) == 0) executionContext.execute(this)
-        else unsafe.getAndAddInt(this, statusOffset, -1)
+        if (activations(again) == 0) true //reschedule
+        else {
+          unsafe.getAndAddInt(this, statusOffset, -1)
+          false // don't reschedule
+        }
+      } else {
+        false // don't reschedule
       }
     }
+
+    val reschedule =
+      try {
+        unscheduleReceiveTimeout()
+        if (!isTerminated(status)) {
+          process()
+          scheduleReceiveTimeout()
+        }
+        postProcess()
+      } catch {
+        case NonFatal(ex) ⇒
+          fail(ex)
+          postProcess()
+        case ie: InterruptedException ⇒
+          fail(ie)
+          if (Debug) println(s"[$thread] $self interrupting due to catching InterruptedException")
+          Thread.currentThread.interrupt()
+          postProcess()
+      }
```

Comment:
Oh, sorry, my bad: `fail(ie)` does not actually execute the Terminate signal, it only enqueues it, so we must schedule.

If execution fails there is not much we can do. We could build in a second try (as in `akka-actor`), but beyond that we can just hope that the actor will receive another message at some point to wake it up again—assuming that the failure to enqueue the cell is transient.

But I think you replied on the wrong line ;-)